File: Python/Fairy_Candy_Swap.py
def fairy_candy_swap():
    alice_sizes = [1, 1, 2]
    bob_sizes = [2, 2]

    suma = sum(alice_sizes)
    sumb = sum(bob_sizes)

    for i in range(len(alice_sizes)):
        for j in range(len(bob_sizes)):
            if suma - alice_sizes[i] + bob_sizes[j] == sumb - bob_sizes[j] + alice_sizes[i]:
                print(f"[{alice_sizes[i]},{bob_sizes[j]}]")
                return

if __name__ == "__main__":
    fairy_candy_swap()

File: Java/Fairy_Candy_Swap.java
void main() {

    int[] aliceSizes = {1,1,2};
    int[] bobSizes = {2,2};

    LinkedList<Integer> arra = new LinkedList<>();
    LinkedList<Integer> arrb = new LinkedList<>();

    for (int num : aliceSizes) {
        arra.add(num);
    }

    for (int num : bobSizes) {
        arrb.add(num);
    }

    int suma = 0;
    int sumb = 0;

    for (int num : arra) {
        suma += num;
    }

    for (int num : arrb) {
        sumb += num;
    }

    for (int i = 0; i < aliceSizes.length; i++) {

        for (int j = 0; j < bobSizes.length; j++) {

            if (
                suma - arra.get(i) + arrb.get(j)
                ==
                sumb - arrb.get(j) + arra.get(i)
            ) {

                int[] answer = {
                    arra.get(i),
                    arrb.get(j)
                };

                System.out.println(
                    "[" + answer[0] + "," + answer[1] + "]"
                );

                return;
            }
        }
    }
}